zip-0227.rst

```diff
@@ -1,8 +1,455 @@
+- Issuing or changing the attributes of a specific Asset should require cryptographic authorization.
+- The Asset identification should be unique (among all shielded pools) and different issuer public keys should not be able to generate the same Asset Identifier.
+- An issuer should be able to issue different Assets in the same transaction. In other words, in a single "issuance bundle", the issuer should be able publish many "issuance actions", potentially creating multiple Custom Assets.
+- Every "issuance action" should contain a ``finalize`` boolean that defines whether the specific Custom Asset can have further tokens issued or not.
+
+
+Specification
+=============
+
+As explained, the issuance protocol allows for a single issuance action to be sent to many receivers, by generating as many output notes as distinct recipients. Furthermore, every bundle can contain many issuance actions.
+
+Issuance Keys and Issuance Authorization Signature Scheme
+---------------------------------------------------------
+
+The ZSA Protocol adds the following three keys to the key components [#protocol-addressesandkeys]_:
+
+1. The issuance master key, denoted as :math:`\mathsf{sk}_{\mathsf{iss}}`, as the name suggests, is the master key that is used to derive the other two keys.
+
+2. The issuance authorizing key is the key that is used to sign the issuance transaction, and is denoted as :math:`\mathsf{isk}`. This key is used to authorize the issuance of a specific Asset Identifier, and is only used by the issuer.
+
+3. The issuance validating key, denoted as :math:`\mathsf{ik}`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific Asset Identifier, and is used by all block chain users (specifically the Asset owners and consensus validators) to associate the Asset in question with the issuer.
+
+The relations between these keys are shown in the following diagram:
+
+.. figure:: zip-0227-key-components-zsa.png
+    :width: 450px
+    :align: center
+    :figclass: align-center
+
+    Diagram of Issuance Key Components for the ZSA Protocol
+
+
+Issuance Authorization Signature Scheme
+```````````````````````````````````````
+
+We define the issuance authorization signature scheme :math:`\mathsf{IssueAuthSig}` similar to :math:`\mathsf{SpendAuthSig}^{\mathsf{Orchard}}`, the Orchard spend authorization signature scheme [#protocol-concretespendauthsig]_. 
+Specifically, we instantiate :math:`\mathsf{IssueAuthSig}` as :math:`\mathsf{RedPallas}` without key re-randomization using generator :math:`\mathcal{P}_{\mathbb{G}} = \mathcal{G}^{\mathsf{Issuance}} := \mathsf{GroupHash}^\mathbb{P}(\texttt{"z.cash:ZSA"}, \texttt{"Issuance"})` where :math:`\mathsf{GroupHash}^\mathbb{P}` is defined as in the Zcash protocol specification [#protocol-concretegrouphashpallasandvesta]_.
```

Comment:
This section seems out of place, I would put after the keys, just before https://deploy-preview-8--zcash-zips-qedit.netlify.app/zip-0227#asset-identifier

Reply:
The reason for defining it first is that IssueAuthSig is used in the [Derivation of issuance authorizing key and issuance validating key](https://deploy-preview-8--zcash-zips-qedit.netlify.app/zip-0227#derivation-of-issuance-authorizing-key-and-issuance-validating-key) section, so it needed to be defined before that.